### DISCUSSION TITLE
Hello. Is it possible to have one 3D object mask out another in AR?

### DISCUSSION THREAD

**Discussion by phatpixels** [google/model-viewer#3920] [Q&A]
I would like to have one 3d object mask another. Say for instance a wall and a door. I would like the wall to mask objects behind it, so the objects are only visible through the door. the walls not visible just act as a mask. See example image. I don't see any examples on this. I have heard of alphaMode, didnt work. Would love to see a working example if this is possible. Many thanks. 

<img width="1001" alt="image" src="https://user-images.githubusercontent.com/38549105/199352494-40e8d7df-aa55-43ae-a645-2992d35aacce.png">

**Comment by elalish** [comment]
We've talked about this before (in the context of a hole in the floor, but same idea). The problem is glTF has no concept of a material like this, though I'd like to see an extension for it. Please propose it to Khronos 3D formats!

**Reply by phatpixels** [reply]
oh i see ok thanks. Here i thought the hole in the floor was something different as it tends to be a one-sided material that occluded itself creating the hole effect, not necessarily masking other objects like i need. I can see now the limitation is with glTF as you point out. I'll be sure to propose it to Khronos. thanks for the tip.
